### PR TITLE
feat(promisify): Adding promisify API to update compat to Node.js v7+

### PR DESCRIPTION
**Pull request by mikeal**

### CODE REVIEW

**Comment by benjaminW78** [comment]
travis integration failed because of old nodejs version targeted....

**Comment by mikeal** [comment]
support for v0.10 and v0.8 should be dropped. minimum supported version should be v4 at this point, critical security vulnerabilities aren't fixed in anything prior any longer.

File: package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
     "test": "node test/node/*.js && zuul test/browser/*.js"
   },
   "dependencies": {
-    "inherits": "2.0.1"
+    "inherits": "2.0.1",
+    "pify": "^3.0.0"
```
Comment (defunctzombie) [inline package.json]: This is not the correct way to specify dependencies. All dependencies should be specified using their exact version. Please follow the existing pattern in a file when making changes.

**Comment by AndersDJohnson** [comment]
@mikeal @defunctzombie This would be especially useful since [Webpack uses this library for the web](https://github.com/webpack/node-libs-browser).

**Comment by wmhilton** [comment]
If we're going with exact version numbers, I vote for hard-pinning pify on version 3.0.0. I've been using pify for about a year, it's my "promisify" shim of choice in all my projects, and I've never had any problem with it. The native `util.promisify` has performance advantages though that I would like to take advantage of on Node, and I anticipate future packages expecting `util.promisify` to be present. If we could push this change through quickly that would be great.

@sindresorhus you haven't touched `pify` since the 3.0.0 release 4 months ago, it's downloaded 218,180 times a day, and it has a whopping ZERO bugs in the issue tracker. Would you agree it's ready for inclusion in this, the official browserify `util` shim?

**Comment by sindresorhus** [comment]
> Would you agree it's ready for inclusion in this, the official browserify util shim?

👍

**Comment by wmhilton** [comment]
Eww... the reason the Travis tests fail is it's running Node 0.10. Barfs at the first "const". Suggestions for how to approach this?

- drop ancient Node support... might cause problems for some people. :(
- transpile 'pify' to ES2000s
- major version bump `node-util`

But all this is moot if this repo is, in fact, a defunct zombie. Pretty quiet on the code committing front since 2014. @substack if we have to fork node-util to get some of these pull requests merged would you be interested in updating browserify to use the maintained fork?

**Comment by CMCDragonkai** [comment]
I'd prefer for this repo to have extra maintainers added, because it owns the `path` name on npm, and so you can easily use this as a polyfill `import ... from 'path';`.

**Comment by wmhilton** [comment]
@mikeal Would you like to update the package.json to use an exact version, since that was @defunctzombie's concern? Also update the Travis file so it passes? Then maybe @goto-bus-stop could merge it and prepare a new release?

**Comment by mikeal** [comment]
I ended up re-working my project to just avoid this problem since the issue has languished for so long. I’m not really interested in diving back into this 6 months later.

**Comment by laino** [comment]
Is this still alive?

**Comment by wmhilton** [comment]
I think it's being worked on here https://github.com/defunctzombie/node-util/pull/25

**Pull request closed without merging**